diff --git a/node_modules/hexo-theme-butterfly/layout/includes/third-party/comments/gitalk.pug b/node_modules/hexo-theme-butterfly/layout/includes/third-party/comments/gitalk.pug
index fccb64d..ca9f066 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/third-party/comments/gitalk.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/third-party/comments/gitalk.pug
@@ -14,7 +14,7 @@ script.
         repo: '!{theme.gitalk.repo}',
         owner: '!{theme.gitalk.owner}',
         admin: ['!{theme.gitalk.admin}'],
-        id: '!{md5(page.path)}',
+        id: '!{pmd5(page.path.split("/").slice(-2)[0])}',
         updateCountCallback: commentCount
       },!{JSON.stringify(theme.gitalk.option)}))
 
diff --git a/node_modules/hexo-theme-butterfly/layout/includes/third-party/comments/waline.pug b/node_modules/hexo-theme-butterfly/layout/includes/third-party/comments/waline.pug
index 06e8078..c9e22f1 100644
--- a/node_modules/hexo-theme-butterfly/layout/includes/third-party/comments/waline.pug
+++ b/node_modules/hexo-theme-butterfly/layout/includes/third-party/comments/waline.pug
@@ -16,7 +16,7 @@ script.
         serverURL: '!{serverURL}',
         pageview: !{lazyload ? false : pageview},
         dark: 'html[data-theme="dark"]',
-        path: window.location.pathname,
+        path: window.location.pathname.split("/").slice(-2)[0],
         comment: !{lazyload ? false : count},
       }, !{JSON.stringify(option)}))
     }
diff --git a/node_modules/hexo-theme-butterfly/scripts/filters/random_cover.js b/node_modules/hexo-theme-butterfly/scripts/filters/random_cover.js
index 47ed4fe..34c0309 100644
--- a/node_modules/hexo-theme-butterfly/scripts/filters/random_cover.js
+++ b/node_modules/hexo-theme-butterfly/scripts/filters/random_cover.js
@@ -15,16 +15,18 @@ hexo.extend.filter.register('before_post_render', function (data) {
     if (cover && cover.indexOf('/') === -1) data.cover = data.path + cover
   }
 
+  console.log("data",data)
+
   if (data.cover === false) {
-    data.randomcover = randomCover()
+    data.randomcover = randomCover(data.title)
     return data
   }
 
-  data.cover = data.cover || randomCover()
+  data.cover = data.cover || randomCover(data.title)
   return data
 })
 
-function randomCover () {
+function randomCover (title) {
   const theme = hexo.theme.config
   let cover
   let num
@@ -34,7 +36,8 @@ function randomCover () {
       cover = theme.cover.default_cover
       return cover
     } else {
-      num = Math.floor(Math.random() * theme.cover.default_cover.length)
+      // num = Math.floor(Math.random() * theme.cover.default_cover.length)
+      num = hashCode(title) % theme.cover.default_cover.length
       cover = theme.cover.default_cover[num]
       return cover
     }
@@ -43,3 +46,13 @@ function randomCover () {
     return cover
   }
 }
+function hashCode(str) {
+  var hash = 0, i, chr;
+  if (str.length === 0) return hash;
+  for (i = 0; i < str.length; i++) {
+    chr   = str.charCodeAt(i);
+    hash  = ((hash << 5) - hash) + chr;
+    hash |= 0; // Convert to 32bit integer
+  }
+  return hash;
+};
diff --git a/node_modules/hexo-theme-butterfly/scripts/helpers/page.js b/node_modules/hexo-theme-butterfly/scripts/helpers/page.js
index aa2b74b..449fe16 100644
--- a/node_modules/hexo-theme-butterfly/scripts/helpers/page.js
+++ b/node_modules/hexo-theme-butterfly/scripts/helpers/page.js
@@ -9,6 +9,7 @@
 
 const { stripHTML, escapeHTML, prettyUrls } = require('hexo-util')
 const crypto = require('crypto')
+const md5 = require('blueimp-md5');
 
 hexo.extend.helper.register('page_description', function () {
   const { config, page } = this
@@ -62,6 +63,10 @@ hexo.extend.helper.register('md5', function (path) {
   return crypto.createHash('md5').update(decodeURI(this.url_for(path))).digest('hex')
 })
 
+hexo.extend.helper.register('pmd5', function (path) {
+  return md5(path)
+})
+
 hexo.extend.helper.register('injectHtml', function (data) {
   let result = ''
   if (!data) return ''
