diff --git a/node_modules/yuque-hexo/lib/Downloader.js b/node_modules/yuque-hexo/lib/Downloader.js
index bf242bb..e2eb47a 100644
--- a/node_modules/yuque-hexo/lib/Downloader.js
+++ b/node_modules/yuque-hexo/lib/Downloader.js
@@ -81,7 +81,7 @@ class Downloader {
    * @return {Promise} queue
    */
   async fetchArticles() {
-    const { client, config, _cachedArticles } = this;
+    const { client, config } = this;
     const articles = await client.getArticles();
     if (!Array.isArray(articles.data)) {
       throw new Error(
@@ -95,6 +95,9 @@ class Downloader {
       )
       .filter(article => (config.onlyPublic ? !!article.public : true))
       .map(article => lodash.pick(article, PICK_PROPERTY));
+
+    this._cachedArticles = this._cachedArticles.filter(cache => realArticles.findIndex((item)=>item.slug === cache.slug) != -1);
+
     const queue = new Queue({ concurrency: config.concurrency });
 
     let article;
@@ -106,6 +109,9 @@ class Downloader {
       return item.slug === article.slug;
     };
     
+
+    const { _cachedArticles } = this;
+
     for (let i = 0; i < realArticles.length; i++) {
       article = realArticles[i];
       cacheIndex = _cachedArticles.findIndex(findIndexFn);
